### PR TITLE
Fixed #632 -- Added a wrapped password hasher to upgrade SHA1 hashes.

### LINKED ISSUE #632: Use a wrapped password hasher to upgrade SHA1 passwords
See https://code.djangoproject.com/ticket/26188

### PR DESCRIPTION
**Pull request by timgraham**

### CODE REVIEW

**Comment by timgraham** [comment]
@apollo13 @dstufft -- I think this is correct. I did a manual login test after an upgrade and it worked.

File: accounts/hashers.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+from django.contrib.auth.hashers import PBKDF2PasswordHasher
+from django.utils.crypto import pbkdf2
+from django.utils.encoding import force_bytes
+
+
+class PBKDF2WrappedSHA1PasswordHasher(PBKDF2PasswordHasher):
+    algorithm = 'pbkdf2_wrapped_sha1'
+
+    def encode_sha1_hash(self, sha1_hash, salt, iterations=None):
```
Comment (apollo13) [inline accounts/hashers.py]: Isn't that just a copy of the original `encode` method?, ie couldn't you just call `super` in `encode` below?

Comment (timgraham) [inline accounts/hashers.py]: Thanks, have refactored to reuse logic from the hashers.

Comment (apollo13) [inline accounts/hashers.py]: That is one way of doing it, although you could just drop the method in it's entirety, that said if you do not want to import the original pbkdf2 hasher in the migration file, this looks good to go…

**Pull request merged**